junos/resource_interface_physical.go

```diff
@@ -72,6 +73,41 @@ func resourceInterfacePhysical() *schema.Resource {
 				Type:     schema.TypeString,
 				Optional: true,
 			},
+			"esi": {
+				Type:     schema.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"mode": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validation.StringInSlice([]string{"all-active", "single-active"}, false),
+						},
+						"auto_derive_lacp": {
+							Type:          schema.TypeBool,
+							Optional:      true,
+							ConflictsWith: []string{"esi.0.identifier"},
+						},
+						"df_election_type": {
+							Type:         schema.TypeString,
+							Optional:     true,
+							ValidateFunc: validation.StringInSlice([]string{"mod", "preference"}, false),
+						},
+						"identifier": {
+							Type:             schema.TypeString,
+							Optional:         true,
+							ConflictsWith:    []string{"esi.0.auto_derive_lacp"},
+							ValidateDiagFunc: validateByteString(10),
+						},
+						"source_bmac": {
+							Type:             schema.TypeString,
+							Optional:         true,
+							ValidateDiagFunc: validateByteString(6),
```

Comment:
Use builtin sdk function
```suggestion
							ValidateFunc: validation.StringMatch(regexp.MustCompile(
								`^([\\d\\w]{2}:){6}[\\d\\w]{2}$`), "bad format or length"),
```